pkg/kubelet/cm/helpers_linux.go

```diff
@@ -169,6 +170,11 @@ func ResourceConfigForPod(pod *v1.Pod, enforceCPULimits bool, cpuPeriod uint64)
 	// build the result
 	result := &ResourceConfig{}
 	if qosClass == v1.PodQOSGuaranteed {
+		// if the pods are in Guaranteed QoS class and cpu manager is enabled and cpu manager policy is static, we should
+		// disable the use of cfs quota at the pod level
+		if utilfeature.DefaultFeatureGate.Enabled(kubefeatures.CPUManager) && (cpuManagerPolicy == string(cpumanager.PolicyStatic)) {
```

Comment:
I do understand the reasoning behind this, but I think this logic should somehow live inside the CPU manager, and not here. We cannot say with 100% certainty that CPU that if a pod spec passes a given validation, it will get assigned to a cpuset, since we don't know the packing of the node here.